openff/toolkit/typing/engines/smirnoff/forcefield.py

```diff
@@ -935,14 +929,39 @@ def _load_smirnoff_data(self, smirnoff_data: dict, allow_cosmetic_attributes=Fal
 
             # Get the parameter types serialization that is not passed to the ParameterHandler constructor.
             ph_class = self._get_parameter_handler_class(parameter_name)
+
             try:
                 infotype = ph_class._INFOTYPE
                 parameter_list_tagname = infotype._ELEMENT_NAME
             except AttributeError:
                 # The ParameterHandler doesn't have ParameterTypes (e.g. ToolkitAM1BCCHandler).
                 parameter_list_dict = {}
+                ignore_keys = []
             else:
                 parameter_list_dict = section_dict.pop(parameter_list_tagname, {})
+                attributes = infotype._get_parameter_attributes()
+                ignore_keys = [
+                    attribute_name
+                    for attribute_name, attribute in attributes.items()
+                    if attribute._unit is None
+                ]
+                # for some reason the angle parameters in VirtualSite
+                # do not have `_unit`, despite being defined with
+                # `unit=unit.degrees`
```

Comment:
I doubt this detail is intentional - could be updated there vs. handling as a special case here?

Reply:
Sorry, that wasn't very clear on my part -- I don't know why this is happening. I've raised #1641 for discussion.